tests/main/interfaces-libvirt/task.yaml

```diff
@@ -0,0 +1,82 @@
+summary: Ensure that the libvirt interface works.
+
+systems: [ubuntu-16.04-64]
+
+summary: |
+    The libvirt interface allows a snap to access the libvirtd socket in order to manage
+    libvirt domains and other resources.
+
+    A snap which defines a libvirt plug must be shown in the interfaces list.
+    The plug must not be autoconnected on install and, as usual, must be able to be
+    reconnected.
+
+    A snap declaring a plug on this interface must be able to create and destroy a domain.
+    The test uses a snap that carries a unikernel built to be run on top of qemu, boot and
+    respond to ping. Once the domain is created, the test checks connectivity to the unikernel.
+
+prepare: |
+    echo "Given libvirt and qemu are installed"
+    sysctl -w net.ipv6.conf.all.disable_ipv6=1
+    trap "sysctl -w net.ipv6.conf.all.disable_ipv6=0" EXIT
+    apt install -y libvirt-bin qemu
+    adduser test libvirtd
+
+    echo "And libvirt is configured to manage /dev/net/tun"
+    systemctl stop libvirtd.service || true
+    echo 'cgroup_device_acl = ["/dev/net/tun", "/dev/random", "/dev/urandom"]' | tee -a /etc/libvirt/qemu.conf
+
+    echo "And the required services up"
+    systemctl start libvirtd.service
+    systemctl start virtlogd.socket
+
+    echo "And a snap declaring a plug on the libvirt interface is installed"
+    snap install --edge test-snapd-libvirt-consumer
+
+    echo "And the required tap interface is in place"
+    ip tuntap add tap100 mode tap
+    ip addr add 10.0.0.1/24 dev tap100
+    ip link set dev tap100 up
+
+restore: |
+    systemctl stop libvirtd.service
+    systemctl stop virtlogd.socket
+    apt remove -y --purge libvirt-bin qemu
```

Comment:
That's likely not enough to undo the above operation. I bet libvirt-bin pulls in a lot of stuff. Should probably at least autoremove.